deso/block.go

```diff
@@ -311,6 +206,151 @@ func (node *Node) GetTransactionsForConvertBlock(block *lib.MsgDeSoBlock) []*typ
 	return transactions
 }
 
+func (node *Node) getOperationsForTransaction(
+	txn *lib.MsgDeSoTxn,
+	block *lib.MsgDeSoBlock,
+	spentUtxos map[lib.UtxoKey]uint64,
+	utxoOpsForTxn []*lib.UtxoOperation,
+) []*types.Operation {
+	if txn.TxnMeta.GetTxnType() == lib.TxnTypeAtomicTxnsWrapper {
+		// There are no operations for the wrapper, but we need to parse
+		// operations for all inner txns.
+		innerTxns := txn.TxnMeta.(*lib.AtomicTxnsWrapperMetadata).Txns
+		ops := []*types.Operation{}
+		var opsForInnerTxns [][]*lib.UtxoOperation
+		for _, utxoOp := range utxoOpsForTxn {
+			if utxoOp.Type == lib.OperationTypeAtomicTxnsWrapper {
+				opsForInnerTxns = utxoOp.AtomicTxnsInnerUtxoOps
+				break
+			}
+		}
+		if len(opsForInnerTxns) != len(innerTxns) {
+			glog.Errorf(
+				"Error: opsForInnerTxns length %v != innerTxns length %v\n", len(opsForInnerTxns), len(innerTxns))
+			return ops
+		}
+		for ii, innerTxn := range innerTxns {
+			ops = append(ops, node.getOperationsForTransaction(innerTxn, block, spentUtxos, opsForInnerTxns[ii])...)
+		}
+		return ops
```

Comment:
Reading this block here made me realize there are a couple of questions we probably have to answer:

1. What is the identifier for an atomic txn? Should we have a synthetic hash that uniquely identifies it so it can be looked up directly? Or should looking it up consist of looking up one of its child txns?
2. Along those lines, I'm trying to think if Rosetta is going to get mad at us for "breaking up" a single txn into multiple txns, which is what we're doing here. But I think it's fine.

I'll send a message in Slack with this thought shortly.

Reply:
It's the same hash as any other transaction! it *just works* lol